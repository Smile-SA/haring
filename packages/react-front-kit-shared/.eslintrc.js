module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
