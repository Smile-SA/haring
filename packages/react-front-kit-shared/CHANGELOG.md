# @smile/react-front-kit-shared

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/react-front-kit/pull/53) [`01ee3ea`](https://github.com/Smile-SA/react-front-kit/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
