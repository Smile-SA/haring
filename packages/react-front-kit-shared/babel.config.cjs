const { babelConfig } = require('test');

module.exports = babelConfig;
