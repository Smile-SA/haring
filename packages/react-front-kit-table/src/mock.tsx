/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
