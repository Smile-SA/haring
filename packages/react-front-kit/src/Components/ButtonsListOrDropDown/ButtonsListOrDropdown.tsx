'use client';
import type { ButtonProps, MenuProps } from '@mantine/core';
import type { IItems } from '@smile/react-front-kit-shared';
import type { ReactElement } from 'react';

import { Button, Menu, createStyles } from '@mantine/core';
import { useUncontrolled } from '@mantine/hooks';

const useStyles = createStyles((theme) => ({
  buttonActive: {
    backgroundColor: theme.colors[theme.primaryColor][8],
    fontWeight: 700,
  },
}));

export interface IButtonsListOrDropdownProps extends MenuProps {
  buttonProps?: ButtonProps;
  current?: string;
  defaultCurrent: string;
  items: IItems<string>;
  menuIfValueGreaterThan?: number;
  onAction?: () => void;
}

export function ButtonsListOrDropdown(
  props: IButtonsListOrDropdownProps,
): ReactElement {
  const {
    buttonProps,
    current,
    menuIfValueGreaterThan = 0,
    defaultCurrent,
    items = [],
    onAction,
    ...MenuProps
  } = props;

  const { classes } = useStyles();

  const [_current, setCurrent] = useUncontrolled({
    defaultValue: defaultCurrent,
    onChange: onAction,
    value: current,
  });

  function handleChange(currentValue: string): void {
    setCurrent(currentValue);
  }

  function getCurrentItem(currentValue: string): string {
    const currentItem = items.filter((item) => item.value === currentValue);
    return currentItem[0]?.value ? currentItem[0].value : currentValue;
  }

  return (
    <>
      {items.length < menuIfValueGreaterThan ? (
        <Button.Group>
          {items.map((item) => {
            return (
              <Button
                className={item.value === _current ? classes.buttonActive : ''}
                {...buttonProps}
                key={item.value}
                onClick={() => handleChange(item.value)}
              >
                {item.content ? item.content : item.label}
              </Button>
            );
          })}
        </Button.Group>
      ) : (
        <Menu {...MenuProps}>
          <Menu.Target>
            <Button {...buttonProps}>{getCurrentItem(_current)}</Button>
          </Menu.Target>
          <Menu.Dropdown>
            {items.map((item) => {
              if (item.value !== _current) {
                return (
                  <Menu.Item
                    key={`${item.value}`}
                    onClick={() => handleChange(item.value)}
                  >
                    {item.content ? item.content : item.label}
                  </Menu.Item>
                );
              }
              return '';
            })}
          </Menu.Dropdown>
        </Menu>
      )}
    </>
  );
}
