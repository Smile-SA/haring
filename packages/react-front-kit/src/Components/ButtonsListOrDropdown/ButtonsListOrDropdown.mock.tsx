export const items = [
  { label: 'FR', value: 'FR' },
  { label: 'EN', value: 'EN' },
  { label: 'ES', value: 'ES' },
];
