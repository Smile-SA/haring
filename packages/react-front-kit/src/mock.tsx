/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
