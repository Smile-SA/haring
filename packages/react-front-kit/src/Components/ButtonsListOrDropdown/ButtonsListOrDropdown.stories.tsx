import type { Meta, StoryObj } from '@storybook/react';

import { ButtonsListOrDropdown as Cmp } from './ButtonsListOrDropdown';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/ButtonsListOrDropdown',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const ButtonsListOrDropdown: IStory = {
  args: {
    defaultCurrent: 'FR',
    displayAll: true,
    items: [
      { label: 'FR', value: 'FR' },
      { label: 'EN', value: 'EN' },
      { label: 'ES', value: 'ES' },
    ],
  },
};
