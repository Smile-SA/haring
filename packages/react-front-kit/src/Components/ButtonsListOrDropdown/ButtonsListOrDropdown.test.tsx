import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { ButtonsListOrDropdown } from './ButtonsListOrDropdown';
import { items } from './ButtonsListOrDropdown.mock';

describe('ButtonsListOrDropdown', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <ButtonsListOrDropdown defaultCurrent="FR" items={items} />,
    );
    expect(container).toMatchSnapshot();
  });
});
