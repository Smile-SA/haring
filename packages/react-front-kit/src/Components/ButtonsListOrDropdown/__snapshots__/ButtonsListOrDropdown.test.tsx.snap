// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsListOrDropdown matches snapshot 1`] = `
<div>
  <button
    aria-controls="mantine-oqv3r3k7h-dropdown"
    aria-expanded="false"
    aria-haspopup="menu"
    class="mantine-UnstyledButton-root mantine-Button-root mantine-1w0871i"
    data-button="true"
    id="mantine-oqv3r3k7h-target"
    type="button"
  >
    <div
      class="mantine-1wpc1xj mantine-Button-inner"
    >
      <span
        class="mantine-1ryt1ht mantine-Button-label"
      />
    </div>
  </button>
</div>
`;
